src/decision/src/subroutines/gate/LineUpWithGate.cpp

```diff
@@ -20,64 +21,71 @@ const gate_detect::GateDetectMsg::ConstPtr& msg) {
     // logic: given the location of the poles, try to put ourselves centred in
     // front
 
+    // send the message
+    geometry_msgs::TwistStamped command;
+
     double x_linear  = 0.0;
     double y_linear  = 0.0;
     double z_linear  = 0.0;
     double x_angular = 0.0;
     double y_angular = 0.0;
     double z_angular = 0.0;
 
-    geometry_msgs::Twist command;
-
-    if (!distance_to_gate_acceptable_) {
-        double averageDistanceToGate;
-
-        averageDistanceToGate =
-        (msg->distanceLeftPole + msg->distanceRightPole +
-         msg->distanceTopPole) /
-        (msg->detectedLeftPole + msg->detectedRightPole + msg->detectedTopPole);
-
-        if (averageDistanceToGate > 7) {
-            if (msg->angleTopPole > 0.25) {
-                command.linear.z = DOWN;
-            } else if (msg->angleTopPole < -0.25) {
-                command.linear.z = UP;
-            }
-            if ((msg->angleLeftPole + msg->angleRightPole) > 0.25) {
-                command.angular.z = -0.25;
-            } else if ((msg->angleLeftPole + msg->angleRightPole) < -0.25) {
-                command.angular.z = 0.25;
-            } else {
-                command.linear.x = FORWARD;
-            }
-        } else {
-            distance_to_gate_acceptable_ = true;
-        }
-    }
-
-    if (!align_top_) {
-        if (msg->angleTopPole > 0.25) {
-            command.linear.z = DOWN;
+    // If we're at an acceptable distance away from the top pole, adjust for
+    // clearance
+    if (msg->detectedTopPole &&
+        abs(msg->distanceTopPole -
```

Comment:
this one particular strikes be as odd. Calling `abs(int)` on a `const float` defined to be positive.

Reply:
`distanceTopPole` doesn't sound like an `int` value, but taking the absolute value makes sense? The distance could be greater or less then the target distance?

Reply:
distanceTopPole is a float, is the standard abs function for ints only?